### ISSUE TITLE
feat: `task list --json` output

### ISSUE THREAD

**Issue by jjjermiah** [prefix-dev/pixi#1434]
### Problem description

This idea is inspired from the `list --json` and provides some ground work for new features regarding the tasks: 

1) one of the features for the vscode extension is creating a **Task Provider** where users can easily run tasks for a project.

    <details open>
    <summary> proof of concept </summary>

    ![image](https://github.com/prefix-dev/pixi/assets/44614774/4872a77e-e8ad-4117-8508-b3e76bb4bf1a)
    
    Ideally, I'd like to add separators in the dropdown menu for environments and features, and the user can start typing either of them and vscode will match on any of those fields. 
    
    </details>

2) As my projects grow, it actually becomes pretty difficult to remember what all my tasks do, and passing as many people work on it, it would be useful to provide context without peers needing to sift through the config file. There was a good idea to https://github.com/prefix-dev/pixi/issues/1113 , and taking that a step further I think it might be helpful to give users a structured understanding of the tasks as well through a tree breakdown. 

    <details>

    <summary> proof of concept `tree`</summary>

    ![image](https://github.com/prefix-dev/pixi/assets/44614774/ce9b3838-1827-4e4c-904f-30bbf6857960)
    
    </details>

I think both of these would greatly benefit from a structured description of the tasks. just fooling around with the code, I started with mirroring the way the packagetojson works with whatever information was stored:

  <details>
  
  <summary> proof of concept </summary>
  
  ![image](https://github.com/prefix-dev/pixi/assets/44614774/1f918a60-249e-4b67-8b76-9e625c52a308)
  
  </details>
  
  I think we would just need to add some information regarding which environment and feature the task belongs to. 
  
  <details>
  
  <summary> high level representation? </summary>

  ```json
  {
    "envs": [
      {
        "<ENVIRONMENT_NAME>": [
          {
            "<FEATURE_NAME>": [
              {
                "<TASK_NAME>": {
                ...
                }
              },
              {
                "<TASK2_NAME>": {
                ...
                }
              }
            ]
          },
          {
            "<FEATURE2_NAME>": [
              {
                "<TASK3_NAME>": {
                ...
                }
              }
            ]
          }
        ]
      },
      {
        "<ENVIRONMENT2_NAME>": [
          {
            "<FEATURE3_NAME>": [
              {
                "<TASK_NAME>": {
                ...
                }
              }
            ]
          }
        ]
      }
    ]
  }
  
  ```
  
  </details>

**Comment by jjjermiah** [comment]
Posting as an idea right now, as I realized this can get very complicated very quickly:

1. in my case, the environments are exclusive for the most part, features dont belong to multiple environments
2. it looks like there are different types of tasks (`Plain`, `Execute`, `Alias`, `Custom`). 
3. I have yet to implement a `pixi.feature.target.<platform>.task` but that might need to be accounted for if those are exclusive to platforms.

in general I think the structured output can just be exhaustive but looking for insight on if there are any glaring flags in this idea

**Comment by jjjermiah** [comment]
closed by #2608
